currency/forexprovider/forexprovider.go

```diff
@@ -24,47 +25,163 @@ func GetAvailableForexProviders() []string {
 
 // NewDefaultFXProvider returns the default forex provider (currencyconverterAPI)
 func NewDefaultFXProvider() *ForexProviders {
-	fxp := new(ForexProviders)
-	currencyC := new(exchangerates.ExchangeRates)
-	currencyC.PrimaryProvider = true
-	currencyC.Enabled = true
-	currencyC.Name = "ExchangeRates"
-	fxp.IFXProviders = append(fxp.IFXProviders, currencyC)
-	return fxp
+	handler := new(ForexProviders)
+	provider := new(exchangerates.ExchangeRates)
+	provider.Setup(base.Settings{
+		PrimaryProvider: true,
+		Enabled:         true,
+		Name:            "ExchangeRates",
+	})
+
+	currencies, _ := provider.GetSupportedCurrencies()
+	providerBase := base.Provider{
+		Provider:            provider,
+		SupportedCurrencies: currencies,
+	}
+
+	handler.FXHandler = base.FXHandler{
+		Primary: providerBase,
+	}
+
+	return handler
 }
 
 // StartFXService starts the forex provider service and returns a pointer to it
-func StartFXService(fxProviders []base.Settings) *ForexProviders {
-	fxp := new(ForexProviders)
+func StartFXService(fxProviders []base.Settings) (*ForexProviders, error) {
+	handler := new(ForexProviders)
+
 	for i := range fxProviders {
 		if fxProviders[i].Name == "CurrencyConverter" && fxProviders[i].Enabled {
-			currencyC := new(currencyconverter.CurrencyConverter)
-			currencyC.Setup(fxProviders[i])
-			fxp.IFXProviders = append(fxp.IFXProviders, currencyC)
+			provider := new(currencyconverter.CurrencyConverter)
```

Comment:
This block here looks to be reused for each fxProvider, consider if you can extract the code to a function, which would minimise code-reuse and simplify changes